### PR TITLE
Adding riscv testing to OCK repo

### PR DESCRIPTION
**Pull request by MaryaSharf**
# Overview

Adding a sample of RISCV examples to OCK to be tested with oneapi.

# Reason for change

A few tests from RISCV examples are added to be tested as a part of new OCK demo package.

### CODE REVIEW

File: examples/applications/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# To compile in a single command:
+#    clang++ -fsycl <file>.cpp -o <file>.fpga_emu
+# CMake executes:
+#    [compile] clang++ -fsycl -o <file>.cpp.o -c <file>.cpp
+#    [link]    clang++ -fsycl <file>.cpp.o -o <file>.fpga_emu
+  message("C++ Compiler: ${CMAKE_CXX_COMPILER}")
+  add_executable(${TARGET} ${SOURCES})
+  set_target_properties(${TARGET} PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS}")
+  set_target_properties(${TARGET} PROPERTIES LINK_FLAGS "${LINK_FLAGS}")
+ # add_custom_target(cpu-gpu DEPENDS ${TARGET})
```
Comment (coldav) [inline examples/applications/CMakeLists.txt:19]: what's this line for?

File: examples/applications/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# CMakeLists.txt
```
Comment (coldav) [inline examples/applications/CMakeLists.txt:1]: Should there be a reference to this in any documentation?

File: examples/applications/CMakeLists.txt (hunk 1 above)
Comment (MaryaSharf) [reply examples/applications/CMakeLists.txt:19]: I missed it, it was probably from older CMakeLists. I'll remove it.

File: examples/applications/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+cmake_minimum_required(VERSION 3.26)
+project(RISCVTests)
+
+set(COMPILE_FLAGS "-fsycl -Wall ${WIN_FLAG}")
+set(LINK_FLAGS "-fsycl")
+
+function(run_riscv_example_with_dpcpp)
+# To compile in a single command:
```
Comment (coldav) [inline examples/applications/CMakeLists.txt:10]: Will we test this in some way or as that part of ock_demo?

Comment (MaryaSharf) [reply examples/applications/CMakeLists.txt:24]: In ock demo. I have already written step, but I changed quite a few things in the networks PR. so was waiting for that to get in to add these tests. I tested them in the normal ock build here ->  https://github.com/codeplaysoftware/oneapi-construction-kit/actions/runs/7570258362/job/20615256925

File: examples/applications/simple-vector-add.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
```
Comment (coldav) [inline examples/applications/simple-vector-add.cpp:19]: No such thing as ComputeCPP any more

**Review by coldav — APPROVED** [review]

**Pull request merged**